src/EmployeesAdministrationApp.java

```diff
@@ -0,0 +1,148 @@
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+import org.json.simple.JSONObject;
+
+public class EmployeesAdministrationApp {
+
+	public static void main(String[] args) {
+
+		switch (chooseAnOptionFromMainMenu()) {
+		case 1:
+			addNewEmployee();
+			break;
+		case 2:
+			viewAllEmployees();
+			break;
+		case 3:
+			searchEmployee();
+			break;
+		case 4:
+			deleteEmployee();
+			break;
+		case 5:
+			updateEmployee();
+			break;
+		case 6:
+			convertEmployee();
+			break;
+		case 7:
+			System.out.println("Bye Bye!");
+			break;
+		}
+	}
+
+	private static void addNewEmployee() {
+		System.out.println("Please select the type of the new employee:"
+				+ "\n1. QA Employee"
+				+ "\n2. Developer Employee"
+				+ "\n3. QA Team Lead Employee " + "\n4. Developer Team Lead Employee" + "\n5. Project Manager");
+			Scanner scan = new Scanner(System.in);
+			int typeOfTheEmployee = scan.nextInt();
+
+			switch (typeOfTheEmployee) {
+			case 1: {
+				QAEmployee qa = new QAEmployee();
```

Comment:
I noticed all the code from the corresponding method has been moved here. this is still duplicated (write a general method that takes as parameters the string to print and the object type as Object. you can check what class instance an object is by using instanceof)
E.g: void generalMethod(String print, Object employee) {<>}
call: generalMethod("Enter the QA employee data", qa);

Reply:
I added a general method which uses the message and the new instance of an class as parameters.